### ISSUE TITLE
Websites are not appearing in the ledger table when you close tab

### ISSUE THREAD

**Issue by kjozwiak** [brave/browser-laptop#11688]
### Description

Sometimes websites are not being added into the ledger even though they've met the conditions outlined in the "Advanced Settings". This doesn't always happen, but I've managed to reproduce it a few times.

### Steps to Reproduce

*Note:* Steps are not 100% and might require a few tries before you see the issue appearing

  1. launch `v0.19.67` rev: b3a897eef1f1d8d6594e2d8e862bf7a874e86e73 
  2. enable "Payments" via `about:preferences#payments`
  3. once the wallet is created, open a new tab and visit a website (I used boardgamegeek.com)
  4. wait for about a minute and switch back to the `about:preferences#payments` tab
  5. you'll notice that sometimes the websites are not appearing under the table

**Actual result:**

Websites are not being added into the ledger table once they've met the conditions outlined in the wallet via "Advanced Settings".

Example GIF: https://drive.google.com/file/d/0B9irDS_rLz-lX1VaajFGT2w5d0U/view?usp=sharing

**Expected result:**

Websites should always be added into the ledger table once they've met the conditions outlined in the wallet via "Advanced Settings", unless they're on the "exclude" list.

**Reproduces how often:** [What percentage of the time does it reproduce?]

Using the above STR/the GIF example... I would say there's about a 30% chance of reproducing the issue.

### Brave Version

**about:brave info:**

> Brave: 0.19.67 
> rev: b3a897eef1f1d8d6594e2d8e862bf7a874e86e73 
> Muon: 4.5.9 
> libchromiumcontent: 62.0.3202.62 

**Reproducible on current live release:**

Yes, I've ran into similar issues before where websites never appear in the ledger table.

**Comment by kjozwiak** [comment]
@NejcZdovc possibly another scenario that hasn't beed addressed in #11274?

**Comment by NejcZdovc** [comment]
ok I see what you did. Crucial step that you forgot to mentions is that you close site tab and not just switch to the preference tab. Because if I switch back to the pref tab I can't reproduce, but if I close tab I can reproduce it.

**Comment by kjozwiak** [comment]
> ok I see what you did. Crucial step that you forgot to mentions is that you close site tab and not just switch to the preference tab. Because if I switch back to the pref tab I can't reproduce, but if I close tab I can reproduce it.

In this particular case, I closed the tab but I've reproduced the issue a bunch of times by simply going through the following STR:

* install a brand new version of brave (usually the latest preview)
* once installed, go into `about:preferences#payments` and enable payments
* once the wallet has been created, open a new tab and visit boardgamegeek.com
* wait for about ~1m40s while scrolling up and down through the website
* switch back to the `about:preferences#payments` and you'll notice sometimes the website didn't appear under the ledger table even though it as met the wallets conditions

I've had this happen a few times while installing a new preview and attempting to test #11675. Once I spent about 1m40s on the website, I switched over to the payments tab and noticed the website never appeared. It doesn't always happen, but I noticed that it's more reproducible using a brand new profile with a new wallet.
